x/group/keeper/keeper_test.go

```diff
@@ -2802,6 +2802,53 @@ func (s *TestSuite) TestLeaveGroup() {
 	}
 }
 
+func (s *TestSuite) TestPruneProposals() {
+	addrs := s.addrs
+	expirationTime := time.Hour * 24 * 15
+	groupID := s.groupID
+	accountAddr := s.groupPolicyAddr
+
+	msgSend := &banktypes.MsgSend{
+		FromAddress: s.groupPolicyAddr.String(),
+		ToAddress:   addrs[0].String(),
+		Amount:      sdk.Coins{sdk.NewInt64Coin("test", 100)},
+	}
+
+	policyReq := &group.MsgCreateGroupPolicy{
+		Admin:   addrs[0].String(),
+		GroupId: groupID,
+	}
+
+	policy := group.NewThresholdDecisionPolicy("100", time.Microsecond, time.Microsecond)
+	err := policyReq.SetDecisionPolicy(policy)
+	s.Require().NoError(err)
+	_, err = s.keeper.CreateGroupPolicy(s.ctx, policyReq)
+	s.Require().NoError(err)
+
+	s.Run("Validate that prune proposal removes expired proposals", func() {
+		req := &group.MsgSubmitProposal{
+			GroupPolicyAddress: accountAddr.String(),
+			Proposers:          []string{addrs[1].String()},
+		}
+		err := req.SetMsgs([]sdk.Msg{msgSend})
+		s.Require().NoError(err)
+		_, err = s.keeper.SubmitProposal(s.ctx, req)
+		s.Require().NoError(err)
+		queryProposal := group.QueryProposalsByGroupPolicyRequest{
+			Address: accountAddr.String(),
+		}
+		prePrune, err := s.keeper.ProposalsByGroupPolicy(s.ctx, &queryProposal)
```

Comment:
Here, instead of getting all the proposals of that group policy, we can just query that one proposal (`s.keeper.Proposal`), using the ID we got above